crates/core/src/kvs/ds.rs

```diff
@@ -279,127 +279,155 @@ impl Datastore {
 
 	#[allow(unused_variables)]
 	pub async fn new_with_clock(path: &str, clock: Option<Arc<SizedClock>>) -> Result<Datastore> {
+		let (flavour, path) = match path.split_once("://").or_else(|| path.split_once(':')) {
+			None if path == "memory" => ("memory", ""),
+			Some((flavour, path)) => (flavour, path),
+			// Validated already in the CLI, should never happen
+			_ => bail!(Error::Unreachable("Provide a valid database path parameter".to_owned())),
+		};
+
+		let path = if Path::new(path).is_absolute() {
+			// if absolute, remove all slashes except one
+			let normalised = format!("/{}", path.trim_start_matches("/"));
+			info!(target: TARGET, "Starting kvs store at {flavour}:{normalised}");
+			info!(target: TARGET, "Note: {flavour}:{normalised} is an absolute path that may require elevated permissions.");
```

Comment:
I get why this note is here but is it really necessary? Even relative paths can resolve to paths that require elevated permissions. I think it might be better to check the error that's returned when trying to create or write to the database folder and give the user a hint if the error is a read-only file system error.